redfish/provider/resource_redfish_manager_reset.go

```diff
@@ -0,0 +1,256 @@
+		return
+	}
+
+	managerID := state.Id.ValueString()
+
+	// Get manager
+	manager, err := getManager(r, state, managerID)
+	if err != nil {
+		resp.Diagnostics.AddError("Error while retrieving manager from redfish API", err.Error())
+		return
+	}
+
+	state.Id = types.StringValue(manager.ID)
+
+	// Save into State
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	tflog.Trace(ctx, "resource_manager_reset read: finished")
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *managerResetResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Get state Data
+	tflog.Trace(ctx, "resource_manager_reset update: started")
+	var state, plan models.RedfishManagerReset
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get plan Data
+	diags = req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	managerID := plan.Id.ValueString()
+
+	// Get Manager
+	manager, err := getManager(r, state, managerID)
+	if err != nil {
+		resp.Diagnostics.AddError("Error while retrieving manager from redfish API", err.Error())
+		return
+	}
+
+	state.Id = types.StringValue(manager.ID)
+	state.ResetType = plan.ResetType
+
+	tflog.Trace(ctx, "resource_manager_reset update: finished state update")
+	// Save into State
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	tflog.Trace(ctx, "resource_manager_reset update: finished")
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (*managerResetResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	tflog.Trace(ctx, "resource_manager_reset delete: started")
+	// Get State Data
+	var state models.RedfishManagerReset
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.State.RemoveResource(ctx)
+	tflog.Trace(ctx, "resource_manager_reset delete: finished")
+}
+
+func getManagerFromCollection(managers []*redfish.Manager, managerID string) (*redfish.Manager, error) {
+	for _, manager := range managers {
+		if manager.ID == managerID {
+			return manager, nil
+		}
+	}
+	return nil, fmt.Errorf("Invalid Manager ID provided")
+}
+
+func getManager(r *managerResetResource, d models.RedfishManagerReset, managerID string) (*redfish.Manager, error) {
+	service, err := NewConfig(r.p, &d.RedfishServer)
+	if err != nil {
+		return nil, err
+	}
+
+	managers, err := service.Managers()
+	if err != nil {
+		return nil, err
+	}
+
+	manager, err := getManagerFromCollection(managers, managerID)
+	if err != nil {
+		return nil, err
+	}
+	return manager, nil
+}
+
+func checkServerStatus(ctx context.Context, endpoint string, interval int, timeout int) error {
+	var err error
+	endpoint = strings.Trim(endpoint, "https://")
```

Comment:
I suggest u use https://pkg.go.dev/net/url#Parse or something equivalent to parse the endpoint string.

Reply:
Dial doesn't require network address in URL format, it expects it in string format.

Reply:
Yes yes, I meant that use https://pkg.go.dev/net/url#URL.Hostname to get the string format.
The idea is that we should not use homegrown parsing of raw URLs when there are full-fledged parsers implemented in the standard libraries.